src/Sylius/Bundle/ApiBundle/Serializer/ProductVariantInputDenormalizer.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Sylius\Bundle\ApiBundle\Serializer;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Sylius\Bundle\ApiBundle\DTO\ChannelPricingInput;
+use Sylius\Bundle\ApiBundle\DTO\ProductVariantInput;
+use Symfony\Component\Serializer\Normalizer\ContextAwareDenormalizerInterface;
+
+class ProductVariantInputDenormalizer implements ContextAwareDenormalizerInterface
+{
+    public function supportsDenormalization($data, string $type, string $format = null, array $context = [])
+    {
+        return $type === ProductVariantInput::class;
+    }
+
+    public function denormalize($data, string $type, string $format = null, array $context = [])
+    {
+        $dto = new ProductVariantInput();
+        if (isset($data['channelPricings'])) $dto->channelPricings = new ArrayCollection($data['channelPricings']);
+        if (isset($data['tracked'])) $dto->tracked = $data['tracked'];
+        if (isset($data['onHand'])) $dto->onHand = $data['onHand'];
+        if (isset($data['weight'])) $dto->weight = $data['weight'];
+        if (isset($data['width'])) $dto->width = $data['width'];
+        if (isset($data['height'])) $dto->height = $data['height'];
+        if (isset($data['depth'])) $dto->depth = $data['depth'];
+        if (isset($data['enabled'])) $dto->enabled = $data['enabled'];
+        if (isset($data['shippingRequired'])) $dto->shippingRequired = $data['shippingRequired'];
+        if (isset($data['optionValues'])) $dto->optionValues = new ArrayCollection($data['optionValues']);
+        if (isset($data['translations'])) $dto->translations = new ArrayCollection($data['translations']);
+
+        return $dto;
```

Comment:
Please read the PSR standard https://www.php-fig.org/psr/psr-12/, and do these code more clarified. Also again, use our php-cs-fixer as I said